pyccel/ast/internals.py

```diff
@@ -182,6 +182,25 @@ class PyccelSymbol(str, Immutable):
     >>> x = PyccelSymbol('x')
     x
     """
+    _is_temp = False
+
+    def __new__(cls, name, is_temp=False):
+        return super().__new__(cls, name)
+
+    def __init__(self, name, is_temp=False):
+        self._is_temp = is_temp
+        super().__init__()
+
+    @property
+    def is_temp(self):
+        """
+        Indicates if the variable created by Pyccel [default value: False]
+        """
+        return self._is_temp
+
+    @is_temp.setter
+    def is_temp(self, is_temp):
+        self._is_temp = is_temp
```

Comment:
As with `Variable` can we avoid the setter? This is more important for `PyccelSymbol` as it should be immutable